operators/test/e2e/volume_test.go

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package e2e
+
+import (
+	"testing"
+
+	"github.com/elastic/cloud-on-k8s/operators/test/e2e/helpers"
+	"github.com/elastic/cloud-on-k8s/operators/test/e2e/stack"
+)
+
+// TestVolumeEmptyDir tests a manual override of the default persistent storage with emptyDir.
+func TestVolumeEmptyDir(t *testing.T) {
+	k := helpers.NewK8sClientOrFatal()
+
+	initStack := stack.NewStackBuilder("test-es-explicit-empty-dir").
+		WithESMasterNodes(1, stack.DefaultResources).
+		WithEmptyDirVolumes()
+
+	helpers.TestStepList{}.
+		WithSteps(stack.InitTestSteps(initStack, k)...).
+		WithSteps(stack.CreationTestSteps(initStack, k)...).
+		WithSteps(stack.ESClusterChecks(initStack.Elasticsearch, k)...).
```

Comment:
I think `ESClusterChecks` is already called from `CreationTestSteps`?
Also since `CreationTestSteps ` calls `K8sStackChecks`, will tests look for a PV in the empty dir scenario?

I was thinking we would have a single `checkVolumes` step that, depending on the provided stack spec, would either look for EmptyDir or PVCs.

Reply:
Oh, yikes, I did not realise how deeply nested the russian dolls were...

Reply:
@sebgl I gave it another try